### PR TITLE
NEO-2117: Table filter button fix and table toolbar button visual fixes

### PR DESCRIPTION
**Pull request by enrique-prado**
[Link to Table example](https://deploy-preview-486--neo-react-library-storybook.netlify.app/?path=/story/components-table--disabled-rows)
[Link to Figma reference](https://www.figma.com/design/o62kJbRQpFq9T0fGwKvCMA/Component-Exploration-FY22---v3?node-id=3313-552263&t=HA9m1SlN5LbBD2sj-0)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

This PR fixes some visual issues with buttons in the Table toolbar.

`@avaya-dux/dux-design`: Verify visuals for the Filter, Refresh and Preferences buttons in the Table toolbar section.

- Filter button should use tertiary variant. (Not filled)
- All buttons should be 32 x 32 pixels.
- Icons inside buttons should be 22 x 22 pixels to match figma designs.

`@avaya-dux/dux-devs`: Visual changes only

- Added new neo-btn-tbl-toolbar class for sizing buttons inside
- large IconButtons should use correct 22px icon size. (Need to move this css code to the css library in future PR)
- Filter button needed to be tertiary.

image

## Summary by CodeRabbit

- **New Features**
	- Introduced two new button styles for improved visibility and usability: large buttons and standardized toolbar buttons.
	- Added column filtering capabilities to the table component for enhanced data management.

- **Enhancements**
	- Improved appearance and functionality of filter buttons in tables by adding size and styling properties.
	- Enhanced the toolbar interface with consistent button styling options.

- **Bug Fixes**
	- Maintained existing functionality while improving button interactions and user interface responsiveness.

### CODE REVIEW

File: src/components/IconButton/IconButton_shim.css [hunk 1]
```diff
@@ -3,6 +3,15 @@
 	top: 5px;
 }
 
+.neo-btn.neo-btn--large {
+	font-size: 22px;
+}
+
+.neo-btn-tbl-toolbar{
```
Comment (yangAtSpoken) [inline src/components/IconButton/IconButton_shim.css:10]: According to BEM naming convention, it should be either .neo-btn--tbl-toolbar or .neo-table__toolbar-btn.

**Review by yangAtSpoken — COMMENTED** [review]
left a comment

**Comment by Justin-Avaya** [comment]
# Design checklist
- [x] Filter button should use tertiary variant. (Not filled)
- [x] All buttons should be 32 x 32 pixels.
- [x] Icons inside buttons should be 22 x 22 pixels to match figma designs.

## Possible approval blocker(s)

**Issue 01**
Horizontal gap between toolbar buttons is not correct. Linked storybook example displays 0px gap between buttons. This should be an 8px gap.

Example(s):


Screenshot 2024-08-06 at 9 10 18 AM

File: src/components/IconButton/IconButton_shim.css (hunk 1 above)
Comment (enrique-prado) [reply src/components/IconButton/IconButton_shim.css]: Good catch, changed it to .neo-table__toolbar-btn

**Review by yangAtSpoken — APPROVED** [review]

**Review by Justin-Avaya — APPROVED** [review]
- [x] Filter button should use tertiary variant. (Not filled)
- [x] All buttons should be 32 x 32 pixels.
- [x] Icons inside buttons should be 22 x 22 pixels to match figma designs.

Approved

**Review by matthewAvaya — COMMENTED** [review]
Approved.

**Pull request merged**